kratos/processes/apply_periodic_boundary_condition_process.cpp

```diff
@@ -0,0 +1,357 @@
+        else
+            KRATOS_ERROR_IF(mTheta != 0.0 && mModulus != 0.0)<<"Both angle of rotation and modulus of translation cannot be specified at the same time. Please check the input"<<std::endl;
+
+        CalculateTransformationMatrix();
+        mIsInitialized = false;
+    }
+
+
+    ApplyPeriodicConditionProcess::~ApplyPeriodicConditionProcess()
+    {
+    }
+
+    /**
+     * @brief  Function to remove the common nodes of slave and master modelparts from the slave modelpart
+     */
+    void ApplyPeriodicConditionProcess::RemoveCommonNodesFromSlaveModelPart()
+    {
+        const int num_slave_nodes = mrSlaveModelPart.NumberOfNodes();
+        const ModelPart::NodeIterator it_slave_node_begin = mrSlaveModelPart.NodesBegin();
+        for(int i_node = 0; i_node<num_slave_nodes; ++i_node)
+        {
+            ModelPart::NodeIterator it_slave_node = it_slave_node_begin;
+            std::advance(it_slave_node, i_node);
+            auto iterator = mrMasterModelPart.Nodes().find(it_slave_node->Id());
+            if(iterator != mrMasterModelPart.NodesEnd()){
+                it_slave_node->Set(TO_ERASE);
+            }
+        }
+        mrSlaveModelPart.RemoveNodes(TO_ERASE);
+    }
+
+    /**
+     * @brief Function initializes the process
+     */
+    void ApplyPeriodicConditionProcess::ExecuteInitialize()
+    {
+        KRATOS_TRY;
+        if (!mIsInitialized)
+        {
+            const ProcessInfoPointerType info = mrMasterModelPart.pGetProcessInfo();
+            const int prob_dim = info->GetValue(DOMAIN_SIZE);
+            // Rotate the master so it goes to the slave
+            if (prob_dim == 2){
+                ApplyConstraintsForPeriodicConditions<2>();
+            }
+            else if (prob_dim == 3){
+                ApplyConstraintsForPeriodicConditions<3>();
+            }
+            mIsInitialized = true;
+        }
+        std::cout<<"Initialization of Periodic conditions finished .. !"<<std::endl;
+
+        KRATOS_CATCH("");
+    }
+
+    /**
+     * @brief Function initializes the solution step
+     */
+    void ApplyPeriodicConditionProcess::ExecuteInitializeSolutionStep()
+    {
+    }
+
+    /**
+     * @brief Function to print the information about this current process
+     */
+    void ApplyPeriodicConditionProcess::PrintInfo(std::ostream& rOStream) const
+    {
+        rOStream <<"ApplyPeriodicConditionProcess Process "<<std::endl;
+    }
+
+    template <int TDim>
+    void ApplyPeriodicConditionProcess::ApplyConstraintsForPeriodicConditions()
+    {
+        const double start_apply = OpenMPUtils::GetCurrentTime();
+        const int num_vars = mParameters["variable_names"].size();
+        BinBasedFastPointLocatorConditions<TDim> bin_based_point_locator(mrMasterModelPart);
+        bin_based_point_locator.UpdateSearchDatabase();
+
+        // for bin based point locator
+		VectorType shape_function_values;
+		const int max_results = 10000;
```

Comment:
Do you have to hardcode this?
What happens if I have a huge problem?

Reply:
Now this is a parameter from json